src/cdk/table/table.ts

```diff
@@ -247,11 +266,50 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
     this._onDestroy.next();
     this._onDestroy.complete();
 
-    if (this.dataSource) {
+    if (this.dataSource instanceof DataSource) {
       this.dataSource.disconnect(this);
     }
   }
 
+  /**
+   * Renders rows based on the table's latest set of data, which was either provided directly as an
+   * input or retrieved through an Observable stream (directly or from a DataSource).
+   * Checks for differences in the data since the last diff to perform only the necessary
+   * changes (add/remove/move rows).
+   *
+   * If the table's data source is a DataSource or Observable, this will be invoked each time the
+   * provided Observable stream emits a new data array. Otherwise if your data is being set
```

Comment:
"Otherwise if your data is being set directory" -> "If the dataSource is an array"

"you will need to call this function..." -> "this function must be manually called to render any changes"